### ISSUE TITLE
Cannot run `make test` using Windows Subsystem for Linux.

### ISSUE THREAD

**Issue by JoeSubbiani** [Mbed-TLS/mbedtls#4632]
### Summary

When running the `make test` in Ubuntu Windows Subsystem for Linux, in the windows file system using Windows Terminal, a stream of error messages occur, seemingly related incorrect EOF.

### System information

Mbed TLS version (number or commit id): Presumably any, tested on 2.6, 2.7 and development branches.
Operating system and version: Windows 10, Using WSL Ubuntu
Configuration (if not default, please attach `config.h`): default
Compiler and options (if you used a pre-built binary, please indicate how you obtained it):  gcc
Additional environment information: Using Windows Terminal to clone repository and run tests. works fine in WSL file system

### Expected behavior

`$ PASSED (96 suites, 11288 tests run)`

### Actual behavior

The following errors + more occur
```
./test_suite_cipher.arc4.datax: 46: depends_on:0: not found
./test_suite_cipher.arc4.datax: 47: 4:exp:0:char*:ARC4-128:int:128:int:32:exp:1: not found
./test_suite_cipher.arc4.datax: 49: ARC4: not found
./test_suite_cipher.arc4.datax: 50: depends_on:0: not found
./test_suite_cipher.arc4.datax: 51: 4:exp:0:char*:ARC4-128:int:128:int:33:exp:1: not found
./test_suite_cipher.arc4.datax: 53: ARC4: not found
./test_suite_cipher.arc4.datax: 54: depends_on:0: not found
./test_suite_cipher.arc4.datax: 55: 4:exp:0:char*:ARC4-128:int:128:int:47:exp:1: not found
./test_suite_cipher.arc4.datax: 57: ARC4: not found
./test_suite_cipher.arc4.datax: 58: depends_on:0: not found
./test_suite_cipher.arc4.datax: 59: 4:exp:0:char*:ARC4-128:int:128:int:48:exp:1: not found
./test_suite_cipher.arc4.datax: 61: ARC4: not found
./test_suite_cipher.arc4.datax: 62: depends_on:0: not found
./test_suite_cipher.arc4.datax: 63: 4:exp:0:char*:ARC4-128:int:128:int:49:exp:1: not found
./test_suite_cipher.arc4.datax: 65: ARC4: not found
./test_suite_cipher.arc4.datax: 66: depends_on:0: not found
./test_suite_cipher.arc4.datax: 67: 7:exp:0:int:128:int:0:int:0:exp:1:int:0:int:0:int:0:int:0: not found
./test_suite_cipher.arc4.datax: 69: ARC4: not found
```

### Steps to reproduce

 - Install WSL, and Ubuntu
 - Use WSL through windows terminal (accessing the windows file system, not the wsl file system)
 - Clone the repository
 - `$ make clean`
 - `$ make`
 - `$ make test`

### Additional information

There will also be errors about windows long paths. This can be solved by:
"Navigate to Local Computer Policy > Computer Configuration > Administrative Templates > System > Filesystem. Double click Enable NTFS long paths. Select Enabled, then click OK."

**Comment by yutotakano** [comment]
I've identified that the error comes from Perl's grep function on line 56 of `run-test-suites.pl` not properly filtering out the `.datax` files on WSL, incorrectly assuming they are executables. 

Upon further investigation, this problem stems from how the files on the Windows filesystem when accessed from WSL are given the default umask of 0777. This is the default behaviour, and to change it you have to remount the drive with the `metadata` option, which is unfortunately, still not very intuitive (e.g. it'll revert your permission modifications if the file is opened from Windows).

A more permanent fix would be to add `.datax` to the exclusion pattern match in `run-test-suites.pl`, not just relying on the executable flag.

**Comment by gilles-peskine-arm** [comment]
I've been thinking that test suite executables should have the `.exe` extension on all platforms. This would simplify several of our build/test scripts. It would also avoid the gotcha that when you switch from branch B1 to branch B2 and then run `make clean`, this doesn't clean test suites that exist in B1 but not in B2, since `make clean` could just remove `*.exe` instead of having to remove only known test suites because test suite executables don't follow a simple pattern.

**Comment by aditya-deshpande-arm** [comment]
Tested again by cloning the repo into the Windows file system, accessing it through WSL and running `make test`. It ran successfully.

**Comment by gilles-peskine-arm** [comment]
Fixed by #6543 and #6611.
